palette/src/oklab.rs

```diff
@@ -0,0 +1,786 @@
+        Self::new(l, a, b, alpha)
+    }
+}
+
+impl<Wp, T> FromColorUnclamped<Oklab<Wp, T>> for Oklab<Wp, T>
+where
+    Wp: WhitePoint,
+    T: FloatComponent,
+{
+    fn from_color_unclamped(color: Self) -> Self {
+        color
+    }
+}
+
+/// This implementation is **only valid if the XYZ white point is D65**.
+impl<Wp, T> FromColorUnclamped<Xyz<Wp, T>> for Oklab<Wp, T>
+where
+    Wp: WhitePoint,
+    T: FloatComponent,
+{
+    fn from_color_unclamped(color: Xyz<Wp, T>) -> Self {
+        let m1 = m1();
+        let m2 = m2();
+
+        let Xyz {
+            x: l, y: m, z: s, ..
+        } = multiply_xyz::<_, Wp, _>(&m1, &color);
+
+        let l_m_s_ = Xyz::new(l.cbrt(), m.cbrt(), s.cbrt());
+
+        let Xyz {
+            x: l, y: a, z: b, ..
+        } = multiply_xyz::<_, Wp, _>(&m2, &l_m_s_);
+
+        Self::new(l, a, b)
+    }
+}
+
+impl<Wp, T: FloatComponent> From<(T, T, T)> for Oklab<Wp, T>
+where
+    Wp: WhitePoint,
+{
+    fn from(components: (T, T, T)) -> Self {
+        Self::from_components(components)
+    }
+}
+
+impl<Wp, T: FloatComponent> Into<(T, T, T)> for Oklab<Wp, T>
+where
+    Wp: WhitePoint,
+{
+    fn into(self) -> (T, T, T) {
+        self.into_components()
+    }
+}
+
+impl<Wp: WhitePoint, T: FloatComponent, A: Component> From<(T, T, T, A)>
+    for Alpha<Oklab<Wp, T>, A>
+{
+    fn from(components: (T, T, T, A)) -> Self {
+        Self::from_components(components)
+    }
+}
+
+impl<Wp: WhitePoint, T: FloatComponent, A: Component> Into<(T, T, T, A)>
+    for Alpha<Oklab<Wp, T>, A>
+{
+    fn into(self) -> (T, T, T, A) {
+        self.into_components()
+    }
+}
+
+impl<Wp, T> Limited for Oklab<Wp, T>
+where
+    Wp: WhitePoint,
+    T: FloatComponent,
+{
+    #[rustfmt::skip]
+    fn is_valid(&self) -> bool {
+        self.l >= from_f64(0.0) && self.l <= from_f64(1.0) &&
+        self.a >= from_f64(-1.0) && self.a <= from_f64(1.0) &&
+        self.b >= from_f64(-1.0) && self.b <= from_f64(1.0)
+    }
+
+    fn clamp(&self) -> Self {
+        let mut c = *self;
+        c.clamp_self();
+        c
+    }
+
+    fn clamp_self(&mut self) {
+        self.l = clamp(self.l, from_f64(0.0), from_f64(1.0));
+        self.a = clamp(self.a, from_f64(-1.0), from_f64(1.0));
+        self.b = clamp(self.b, from_f64(-1.0), from_f64(1.0));
+    }
+}
+
+impl<Wp, T> Mix for Oklab<Wp, T>
+where
+    Wp: WhitePoint,
+    T: FloatComponent,
+{
+    type Scalar = T;
+
+    fn mix(&self, other: &Self, factor: T) -> Self {
+        let factor = clamp(factor, T::zero(), T::one());
+
+        Self::new(
+            self.l + factor * (other.l - self.l),
+            self.a + factor * (other.a - self.a),
+            self.b + factor * (other.b - self.b),
+        )
+    }
+}
+
+impl<Wp, T> Shade for Oklab<Wp, T>
+where
+    Wp: WhitePoint,
+    T: FloatComponent,
+{
+    type Scalar = T;
+
+    fn lighten(&self, amount: T) -> Self {
+        Self::new(self.l + amount * from_f64(1.0), self.a, self.b)
```

Comment:
You can remove `* from_f64(1.0)`, since it shouldn't have an effect.

Reply:
Good catch!